src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12906,5 +12959,20 @@ private new struct ResetPoint
             node = this.AddTrailingSkippedSyntax(node, trailingTrash.Node);
             return node;
         }
+
+        private SyntaxToken MergeTokens(SyntaxToken s1, SyntaxToken s2, SyntaxKind kind)
+        {
+            if (s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0)
+            {
+                s1 = SyntaxFactory.Token(s1.GetLeadingTrivia(), kind, s2.GetTrailingTrivia());
+            }
+            else
+            {
+                s2 = this.AddError(s2, ErrorCode.ERR_InvalidExprTerm, this.CurrentToken.Text);
```

Comment:
>this.CurrentToken.Text [](start = 70, length = 22)

Referencing `this.CurrentToken` seems surprising. Should this be `s2` instead?

Reply:
You're absolutely right, changed.